src/layers/domain/repository/repository/v2.py

```diff
@@ -20,7 +24,44 @@
 ModelType = TypeVar("ModelType", bound=AggregateRoot)
 T = TypeVar("T")
 BATCH_SIZE = 100
-MAX_BATCH_WRITE_SIZE = 25
+MAX_BATCH_WRITE_SIZE = 10
+RETRY_ERRORS = [
+    "ProvisionedThroughputExceededException",
+    "ThrottlingException",
+    "InternalServerError",
+]
+
+
+def exponential_backoff_with_jitter(
+    n_retries, base_delay=0.1, min_delay=0.05, max_delay=5
+):
+    """Calculate the delay with exponential backoff and jitter."""
+    delay = min(base_delay * (2**n_retries), max_delay)
+    return random.uniform(min_delay, delay)
+
+
+def retry_with_jitter(max_retries=5, error=ClientError):
```

Comment:
Is this jitter a way to handle the issues you mentioned around the DB taking too much load and potentially failing?

Reply:
yeah if two or more processes fail at the same time it means they won't all restart at exactly the same time

Reply:
jitter retrying is a strategy from my web scraping days